src/genie/libs/parser/iosxe/show_dmvpn.py

```diff
@@ -0,0 +1,124 @@
+# Metaparser
+import re
+from genie.metaparser import MetaParser
+from genie.metaparser.util.schemaengine import Any, Or, Optional
+
+
+# ==============================
+# Schema for 'show dmvpn'
+# ==============================
+class ShowDmvpnSchema(MetaParser):
+    """
+    Schema for 'show dmvpn'
+    Schema for 'show dmvpn interface <WORD>'
+    """
+
+# These are the key-value pairs to add to the parsed dictionary
+    schema = {
```

Comment:
This is a very good schema but I think that 
` schema = {
         'dmvpn': {
             Any(): {
                 'total_peers': str,
                 'type': str,
                 'peers': {
                    'ipv4':{
                        Any(): {
                            'tunnel_addr': str,
                            'state': str,
                            'time': str,
                            'attrb': str,
                            'ent': str
                            },
                        },
                    },
                },
            }
        }`

Reply:
@kamyarz1991 
thanks for the suggestion. seams fair to add ipv4 key.

ipv6 has a diffrent output... :cold_sweat: 
example can be faind her 
https://www.cisco.com/c/en/us/td/docs/ios-xml/ios/sec_conn_dmvpn/configuration/15-s/sec-conn-dmvpn-15-s-book/ip6-dmvpn.html#GUID-C0C976BF-16C0-42DE-B9B6-1A74E9716543

need to add more test

Reply:
Thanks for the update, Everything looks good and ready to go, just one small thing, before we merge this. Can you please add the ipv6 unit test as well and make sure it works in this code. When thats done I think we are good to merge, thanks

Reply:
Also, Can you please in this directory geneparser/tests’ dir run, this python -m unittest command and make sure that all the unittests are passing.
Thank You

Reply:
@KamyarZiabari All unittest test is Okey

```
Ran 3649 tests in 6.978s

OK
```

Reply:
@kamyarz1991 / @KamyarZiabari, Regarding the IPv6 unittest, I don't have any console output with IPv6 to test the parser agings. I thing is the best way i wait for a issue with a ipv6 output and then fix the parser if it is necessary.

Reply:
`Device# show dmvpn ipv6 detail

Legend: Attrb --> S - Static, D - Dynamic, I - Incomplete
        N - NATed, L - Local, X - No Socket
        # Ent --> Number of NHRP entries with same NBMA peer
        NHS Status: E --> Expecting Replies, R --> Responding
        UpDn Time --> Up or Down Time for a Tunnel
==========================================================================

Interface Tunnel1 is up/up, Addr. is 10.0.0.3, VRF "" 
   Tunnel Src./Dest. addr: 192.169.2.9/MGRE, Tunnel VRF ""
   Protocol/Transport: "multi-GRE/IP", Protect "test_profile" 
Type:Hub, Total NBMA Peers (v4/v6): 2
    1.Peer NBMA Address: 192.169.2.10
        Tunnel IPv6 Address: 2001::4
        IPv6 Target Network: 2001::4/128
        # Ent: 2, Status: UP, UpDn Time: 00:01:51, Cache Attrib: D
Type:Hub, Total NBMA Peers (v4/v6): 2
    2.Peer NBMA Address: 192.169.2.10
        Tunnel IPv6 Address: 2001::4
        IPv6 Target Network: FE80::2/128
        # Ent: 0, Status: UP, UpDn Time: 00:01:51, Cache Attrib: D
Type:Hub, Total NBMA Peers (v4/v6): 2
    3.Peer NBMA Address: 192.169.2.11
Tunnel IPv6 Address: 2001::5
        IPv6 Target Network: 2001::5/128
        # Ent: 2, Status: UP, UpDn Time: 00:26:38, Cache Attrib: D
Type:Hub, Total NBMA Peers (v4/v6): 2
    4.Peer NBMA Address: 192.169.2.11
        Tunnel IPv6 Address: 2001::5
        IPv6 Target Network: FE80::3/128
        # Ent: 0, Status: UP, UpDn Time: 00:26:38, Cache Attrib: D
Pending DMVPN Sessions:

Interface: Tunnel1
  IKE SA: local 192.169.2.9/500 remote 192.169.2.10/500 Active 
  Crypto Session Status: UP-ACTIVE     
  fvrf: (none), Phase1_id: 192.169.2.10
  IPSEC FLOW: permit 47 host 192.169.2.9 host 192.169.2.10 
        Active SAs: 2, origin: crypto map
   Outbound SPI : 0x BB0ED02, transform : esp-aes esp-sha-hmac 
    Socket State: Open

Interface: Tunnel1
  IKE SA: local 192.169.2.9/500 remote 192.169.2.11/500 Active 
  Crypto Session Status: UP-ACTIVE     
  fvrf: (none), Phase1_id: 192.169.2.11
  IPSEC FLOW: permit 47 host 192.169.2.9 host 192.169.2.11 
        Active SAs: 2, origin: crypto map
   Outbound SPI : 0xB79B277B, transform : esp-aes esp-sha-hmac 
    Socket State: Open`

Reply:
`
Device# show dmvpn ipv6 detail

Legend: Attrb --> S - Static, D - Dynamic, I - Incomplete
        N - NATed, L - Local, X - No Socket
        # Ent --> Number of NHRP entries with same NBMA peer
        NHS Status: E --> Expecting Replies, R --> Responding
        UpDn Time --> Up or Down Time for a Tunnel
==========================================================================

Interface Tunnel1 is up/up, Addr. is 10.0.0.3, VRF "" 
   Tunnel Src./Dest. addr: 192.169.2.9/MGRE, Tunnel VRF ""
   Protocol/Transport: "multi-GRE/IP", Protect "test_profile" 
Type:Hub, Total NBMA Peers (v4/v6): 2
    1.Peer NBMA Address: 192.169.2.10
        Tunnel IPv6 Address: 2001::4
        IPv6 Target Network: 2001::4/128
        # Ent: 2, Status: UP, UpDn Time: 00:01:51, Cache Attrib: D
Type:Hub, Total NBMA Peers (v4/v6): 2
    2.Peer NBMA Address: 192.169.2.10
        Tunnel IPv6 Address: 2001::4
        IPv6 Target Network: FE80::2/128
        # Ent: 0, Status: UP, UpDn Time: 00:01:51, Cache Attrib: D
Type:Hub, Total NBMA Peers (v4/v6): 2
    3.Peer NBMA Address: 192.169.2.11
Tunnel IPv6 Address: 2001::5
        IPv6 Target Network: 2001::5/128
        # Ent: 2, Status: UP, UpDn Time: 00:26:38, Cache Attrib: D
Type:Hub, Total NBMA Peers (v4/v6): 2
    4.Peer NBMA Address: 192.169.2.11
        Tunnel IPv6 Address: 2001::5
        IPv6 Target Network: FE80::3/128
        # Ent: 0, Status: UP, UpDn Time: 00:26:38, Cache Attrib: D
Pending DMVPN Sessions:

Interface: Tunnel1
  IKE SA: local 192.169.2.9/500 remote 192.169.2.10/500 Active 
  Crypto Session Status: UP-ACTIVE     
  fvrf: (none), Phase1_id: 192.169.2.10
  IPSEC FLOW: permit 47 host 192.169.2.9 host 192.169.2.10 
        Active SAs: 2, origin: crypto map
   Outbound SPI : 0x BB0ED02, transform : esp-aes esp-sha-hmac 
    Socket State: Open

Interface: Tunnel1
  IKE SA: local 192.169.2.9/500 remote 192.169.2.11/500 Active 
  Crypto Session Status: UP-ACTIVE     
  fvrf: (none), Phase1_id: 192.169.2.11
  IPSEC FLOW: permit 47 host 192.169.2.9 host 192.169.2.11 
        Active SAs: 2, origin: crypto map
   Outbound SPI : 0xB79B277B, transform : esp-aes esp-sha-hmac 
    Socket State: Open
`

Reply:
@urskog84
There is one in the above link that you sent under show dmvpn ipv6 detail. By adding that, you need to update the list of commands that are being covered. I will comment separately on that. 
If this is too much work I totally understand, just let me know and I take on from here.
Thanks

Reply:
The Output of with or without "detail" is diffrent.

**without detail**

```
Legend: Attrb --> S - Static, D - Dynamic, I - Incomplete
        N - NATed, L - Local, X - No Socket
        T1 - Route Installed, T2 - Nexthop-override
        C - CTS Capable
        # Ent --> Number of NHRP entries with same NBMA peer
        NHS Status: E --> Expecting Replies, R --> Responding, W --> Waiting
        UpDn Time --> Up or Down Time for a Tunnel
==========================================================================

Interface: Tunnel78, IPv4 NHRP Details
Type:Spoke, NHRP Peers:2,

 # Ent  Peer NBMA Addr Peer Tunnel Add State  UpDn Tm Attrb
 ----- --------------- --------------- ----- -------- -----
     1 172.29.0.1          172.30.78.1   IKE     2w3d     S
     1 172.29.0.2          172.30.78.2    UP     3w4d     S
```
**with detail**
```
tstiprtr001#show dmvpn interface tunnel 78 detail
Legend: Attrb --> S - Static, D - Dynamic, I - Incomplete
        N - NATed, L - Local, X - No Socket
        T1 - Route Installed, T2 - Nexthop-override
        C - CTS Capable
        # Ent --> Number of NHRP entries with same NBMA peer
        NHS Status: E --> Expecting Replies, R --> Responding, W --> Waiting
        UpDn Time --> Up or Down Time for a Tunnel
==========================================================================

Interface Tunnel78 is up/up, Addr. is 172.30.78.253, VRF "FOTO"
   Tunnel Src./Dest. addr: 172.29.255.1/MGRE, Tunnel VRF "FRONTDOOR"
   Protocol/Transport: "multi-GRE/IP", Protect "MIGV"
   Interface State Control: Disabled
   nhrp event-publisher : Disabled

IPv4 NHS:
172.30.78.1  RE priority = 0 cluster = 0
172.30.78.2  RE priority = 0 cluster = 0
Type:Spoke, Total NBMA Peers (v4/v6): 2

# Ent  Peer NBMA Addr Peer Tunnel Add State  UpDn Tm Attrb    Target Network
----- --------------- --------------- ----- -------- ----- -----------------
    1 172.29.0.1          172.30.78.1   IKE     2w3d     S     172.30.78.1/32 (FOTO)
    1 172.29.0.2          172.30.78.2    UP     3w4d     S     172.30.78.2/32 (FOTO)


Crypto Session Details:
--------------------------------------------------------------------------------

Interface: Tunnel72 Tunnel74 Tunnel76 Tunnel78 Tunnel80 Tunnel82 Tunnel88 Tunnel96
Session: [0x7FB25B8010B8]
  Crypto Session Status: UP-NO-IKE
  fvrf: FRONTDOOR,        IPSEC FLOW: permit 47 host 172.29.255.1 host 172.29.0.1
        Active SAs: 2, origin: crypto map
        Inbound:  #pkts dec'ed 112981987 drop 0 life (KB/Sec) 4595846/977
        Outbound: #pkts enc'ed 120032386 drop 0 life (KB/Sec) 4578683/977
   Outbound SPI : 0xB3F099BD, transform : esp-256-aes esp-sha-hmac
    Socket State: Open

Interface: Tunnel72 Tunnel74 Tunnel76 Tunnel78 Tunnel80 Tunnel82 Tunnel88 Tunnel96
Session: [0x7FB25B801538]
  Crypto Session Status: UP-NO-IKE
  fvrf: FRONTDOOR,        IPSEC FLOW: permit 47 host 172.29.255.1 host 172.29.0.2
        Active SAs: 2, origin: crypto map
        Inbound:  #pkts dec'ed 23297424 drop 0 life (KB/Sec) 4606474/2230
        Outbound: #pkts enc'ed 20983259 drop 0 life (KB/Sec) 4606930/2230
   Outbound SPI : 0x381947BA, transform : esp-256-aes esp-sha-hmac
    Socket State: Open

```